### PR TITLE
node: add sui watcher for devnet and testnet only

### PR DESCRIPTION
**Pull request by panoel**
This pulls the SUI guardian watcher from the sui/integration_v2 branch up and only allows it to run in devnet and testnet.

### CODE REVIEW

File: node/pkg/watchers/sui/watcher.go [hunk 1]
```diff
@@ -231,189 +249,221 @@ func (e *Watcher) inspectBody(logger *zap.Logger, body SuiResult) error {
...
+				var res SuiEventMsg
+				err = json.Unmarshal(msg, &res)
+				if err != nil {
+					logger.Error("Failed to unmarshal SuiEventMsg", zap.String("body", string(msg)), zap.Error(err))
+					return fmt.Errorf("Failed to unmarshal SuiEventMsg, body: %s, error: %w", string(msg), err)
+				}
+				if res.Error != nil {
+					return fmt.Errorf("Bad SuiEventMsg, body: %s, error: %w", string(msg), err)
+				}
+				logger.Info("SUI result message", zap.String("message", string(msg)), zap.Any("event", res))
```
Comment (bruce-riley) [inline node/pkg/watchers/sui/watcher.go:338]: Need to remember to change these things to debugs before mainnet.

**Review by bruce-riley — APPROVED** [review]

**Review by evan-gray — CHANGES_REQUESTED** [review]
Seeing Bruce's comments, let's make those changes first 🙂

File: node/pkg/watchers/sui/watcher.go (hunk 1 above)
Comment (panoel) [reply node/pkg/watchers/sui/watcher.go:338]: Done

File: node/cmd/guardiand/node.go [hunk 2]
```diff
@@ -1362,12 +1358,16 @@ func runNode(cmd *cobra.Command, args []string) {
...
 
 		if shouldStart(suiRPC) {
-			logger.Info("Starting Sui watcher")
-			common.MustRegisterReadinessSyncing(vaa.ChainIDSui)
-			chainObsvReqC[vaa.ChainIDSui] = make(chan *gossipv1.ObservationRequest, observationRequestBufferSize)
-			if err := supervisor.Run(ctx, "suiwatch",
-				sui.NewWatcher(*suiRPC, *suiWS, *suiAccount, *suiPackage, *unsafeDevMode, chainMsgC[vaa.ChainIDSui], chainObsvReqC[vaa.ChainIDSui]).Run); err != nil {
-				return err
+			if !*unsafeDevMode && !*testnetMode {
+				logger.Error("Can only start Sui watcher in devnet or testnet")
```
[the remark this answers is not in the document]
Comment (bruce-riley) [reply node/cmd/guardiand/node.go:1362]: I think it should be fatal if they screw up their mainnet config. They should have to fix it.

There are numerous cases of node.go using logger.Fatal(), but returning an error would work too, if you prefer.

[the remark this answers is not in the document]
Comment (panoel) [reply node/cmd/guardiand/node.go:1362]: Changed.

**Review by bruce-riley — APPROVED** [review]

**Pull request merged**